pycqed/measurement/waveform_control/pulse_library.py

```diff
@@ -104,11 +105,9 @@ def chan_wf(self, channel, tvals):
             return np.zeros_like(tvals)
 
     def hashables(self, tstart, channel):
-        if channel not in self.channels:
-            return []
-        if self.pulse_off:
-            return ['Offpulse', self.algorithm_time() - tstart, self.length]
-        hashlist = [type(self), self.algorithm_time() - tstart]
+        hashlist = self.common_hashables(tstart, channel)
```

Comment:
Is this related to Cryoscope?

Reply:
It adds the truncation parameters to the hashables, see definition of common_hashables. This is related to introducing the truncation feature in pulse, which is related to cryoscope. Since we now have the function common_hashables, other parameters could be moved into that function as well instead of replicating the code in every pulse.